packages/public/server/src/module/Entity/src/Controller/RepositoryController.php

```diff
@@ -85,11 +85,12 @@ public function addLegacyRevisionAction()
         if ($this->getRequest()->isPost()) {
             $form->setData($this->getRequest()->getPost());
             if ($form->isValid()) {
-                return $this->handleAddRevisionPost(
+                $redirectUrl = $this->handleAddRevisionPost(
```

Comment:
Pretty sure that this won't work for Edtr.io stuff since that expected the redirect url in the response body (instead of redirecting directly), see: https://github.com/serlo/serlo.org/blob/974c9e5279d5e209015b2b05d5374d7c1dfc5f69/packages/public/client/src/editor/editor/index.tsx#L54-L62.

Reply:
This is for the old editor `function addLegacyRevisionAction()`. At https://github.com/serlo/serlo.org/blob/c0739c6fb060583f58e5b874281b23df575f443f/packages/public/server/src/module/Entity/src/Controller/RepositoryController.php#L132-L142 you have the code for edtr.io

Reply:
Check, `addLegacyRevisionAction` is always a normal HttpRequest while `addRevisionAction` is always a XMLHttpRequest, so the check is no longer needed.